### PR TITLE
[Feature] upload report command

### PR DESCRIPTION
**Pull request by qrtt1**
**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**

Feature added

**What this PR does / why we need it**:

Support upload-report to the cloud

**Which issue(s) this PR fixes**:

sc-28411

**Configuration**

In `~/.piperider/profile.yml` added:

```yaml
api_service: https://staging.cloud.piperider.io/
api_token: <the api token>
```

* api_service has default value: `https://cloud.piperider.io/`
  * local dev: `https://127.0.0.1:3030/`
  * staging: `https://staging.cloud.piperider.io/`

Get the token from the user dashboard:

image

**Commands**

* Add cloud section to `diagnose`
image

* A simple upload command
```
$ piperider upload-report --run /Users/qrtt1/temp/piperider/.piperider/aaaa/b/outputs/latest/run.json
{'filename': '9baa1ca4940c4c00b172588a543ee3b0.json', 'success': True}
```

### CODE REVIEW

**Comment by popcornylu** [comment]
In the command description

> Upload a report to the PipeRider Cloud.

Could we provide a hind to isay it is experimental feature?

> (Experimental) Upload a report to the PipeRider Cloud.

**Comment by kentwelcome** [comment]
By default, we will hidden the `cloud` command in the help description.
Enable to show those beta command by Environment Variable `PIPERIDER_BETA=true`
CleanShot 2022-09-13 at 12 26 59

**Comment by kentwelcome** [comment]
Users should use `piperider cloud login` to login PipeRider cloud first.
Once users execute `login` command, they will receive an email with a magic link.
By accessing that link, they can get the API token and just paste it back to the terminal.
CleanShot 2022-09-13 at 12 30 16

**Comment by kentwelcome** [comment]
Once users complete the login process. Executing the `login` command will show the user's email account.
CleanShot 2022-09-13 at 12 33 01

**Comment by kentwelcome** [comment]
When executing `upload-report` command, we will ask the user to pick a report to upload.
CleanShot 2022-09-13 at 14 45 15

**Comment by kentwelcome** [comment]
The `cloud` will be hidden by default. It only shows when setting the environment variable `PIPERIDER_BETA=true`.

File: piperider_cli/cli.py [hunk 1]
```diff
@@ -220,3 +221,40 @@ def compare_reports(**kwargs):
...
+
+@cloud.command(short_help='Upload a report to the PipeRider Cloud.', beta=True, cls=TrackCommand)
+@click.option('--run', type=click.Path(exists=True), help='Specify the raw result file.')
+@click.option('--report-dir', default=None, type=click.STRING, help='Use a different report directory.')
+@click.option('--datasource', default=None, type=click.STRING, metavar='DATASOURCE_NAME',
+              help='Specify the datasource.')
+@add_options(debug_option)
+def upload_report(**kwargs):
+    """
+    Upload a single run report to PipeRider Cloud
```
Comment (ctiml) [inline piperider_cli/cli.py:240]: I suggest we should remove `single`

Comment (kentwelcome) [reply piperider_cli/cli.py:240]: I will change to use multiple select to let users upload multiple files.

Comment (ctiml) [reply piperider_cli/cli.py:240]: Sounds great!

File: piperider_cli/cloud_connector.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    response = piperider_cloud.magic_login(account)
+    if response is None or response.get('success') is False:
+        console.print('[[red]Error[/red]] Login failed. Please try again.')
+        return False
+
+    if response.get('link'):
+        webbrowser.open(response.get('link'))
+
+    console.print('Please paste the api token from magic link. The link had be sent to your email address.')
```
Comment (ctiml) [inline piperider_cli/cloud_connector.py:35]: Suggestion: `The link had be sent to your email address` -> `The link has been sent to your email address`

File: piperider_cli/cli.py (hunk 1 above)
Comment (kentwelcome) [reply piperider_cli/cli.py:240]: Now, users can select multiple reports to upload. Also refactor the report selection function used by compare-reports.

**Review by ctiml — APPROVED** [review]

**Pull request merged**